### PR TITLE
User could insert a subscription email address

### PR DESCRIPTION
**Pull request by lbai001**

### CODE REVIEW

File: components/DashboardProfile/index.js [hunk 1]
```diff
@@ -10,33 +10,57 @@ class DashboardProfile extends React.Component {
...
+          });
+        } else {
+          this.setState({ disabled: true });
+          swal(
+            'Please enter a valid email address or leave empty',
+            null,
+            'error'
+          );
+        }
+      }
```
Comment (RobAWilkinson) [inline components/DashboardProfile/index.js]: extract this out into a `if(e.target.value != '') { switch(userField) { case 'userEmail':  } }`

Comment (RobAWilkinson) [reply components/DashboardProfile/index.js]: could also set the message you pass to sweetalert in a variable and just change it in the else statements of the different fields

File: components/DashboardProfile/index.js [hunk 2]
```diff
@@ -29,34 +36,54 @@ class DashboardProfile extends React.Component {
   handleEdit = () => {
     this.setState({ edit: true });
   };
+  validateEmail(email) {
+    var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (RobAWilkinson) [inline components/DashboardProfile/index.js:40]: `const`

File: components/DashboardProfile/index.js [hunk 3]
```diff
@@ -29,34 +36,54 @@ class DashboardProfile extends React.Component {
...
+            this.setState({ disabled: false });
+          }
+        }
+      default:
+        this.setState({
+          user_metadata: {
+            ...this.state.user_metadata,
+            [field]: e.target.value
+          }
+        });
```
Comment (RobAWilkinson) [inline components/DashboardProfile/index.js:68]: is it possible to remove this duplication of:
```
          this.setState({
            disabled: false,
            user_metadata: {
              ...this.state.user_metadata,
              userEmail: e.target.value
            }
          });
``` 

perhaps refactor the first case to `if(!this.validateEmail)`? 

And use the fact that a switch statement will "fall through" unless you use a `break` [here](https://stackoverflow.com/questions/13207927/switch-statement-multiple-cases-in-javascript#13207939)

File: components/DashboardProfile/index.js [hunk 4]
```diff
@@ -9,11 +9,18 @@ import { Button } from 'react-bootstrap';
...
     super(props);
+    this.messages = {
+      emailValidation: 'Please enter a valid email address or leave empty',
+      profileSaved: 'Profile saved!',
+      enterNickname: 'Please edit your profile to set a nickname'
+    };
     this.state = {
+      disabled: false,
       edit: false,
       user_metadata: {
```
Comment (RobAWilkinson) [inline components/DashboardProfile/index.js:20]: why not opt for `this.state.profile` just to keep things separate. 

React code: `state.profile` `props.profile` etc. etc. 

Auth0 code: `user_metadata`

**Review by RobAWilkinson — CHANGES_REQUESTED** [review]
Left some comments, it would also be great to refactor the `this.props.foo` in render statements into `const { foo } = this.props`.

This can improve readability.

File: components/DashboardProfile/index.js (hunk 4 above)
Comment (lbai001) [reply components/DashboardProfile/index.js:20]: check

File: components/DashboardProfile/index.js (hunk 2 above)
Comment (lbai001) [reply components/DashboardProfile/index.js:40]: check

File: components/DashboardProfile/index.js (hunk 3 above)
Comment (lbai001) [reply components/DashboardProfile/index.js:68]: check

**Pull request closed without merging**